okhttp-dnsoverhttps/src/main/kotlin/okhttp3/dnsoverhttps/InetAddressUtil.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.dnsoverhttps
+
+import java.lang.IllegalArgumentException
+import java.lang.NumberFormatException
+import java.lang.StringBuilder
+import java.net.InetAddress
+import java.net.UnknownHostException
+import java.util.*
+
+object InetAddressUtil {
```

Comment:
This looks like a lot of supporting code.  Are there existing methods that do this, or would a simple regex as a first parse be simpler (\d+\.\d+...) or (\d+:...)?

Reply:
I implement the InetAddressUtil kotlin version refer to the com.google.common.net.InetAddresses class. I just concern about that a simple regex may cause some bugs due to the complex format of IPv4/IPv6. A common parse algorithm would be better.

Reply:
I also updated the copyright just same as the raw implement file.